test/rule-matches/label-content-name-mismatch-matches.js

```diff
@@ -0,0 +1,144 @@
+		var vNode = queryFixture(
+			'<div role="button" id="target" aria-labelledby="foo">some content</div>' +
+				'<div id="foo"></div>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns false if element has empty accessible name (`aria-labelledby`) because idref does not exist', function() {
+		var vNode = queryFixture(
+			'<div role="button" id="target" aria-labelledby="doesNotExist">some content</div>' +
+				'<div id="idExists">Right Label</div>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns true if element has accessible name (`aria-labelledby`) - multiple refs', function() {
+		var vNode = queryFixture(
+			'<div role="button" id="target" aria-labelledby="bar baz foo">some content</div>' +
+				'<div id="foo">Foo</div>' +
+				'<div id="bar">Bar</div>' +
+				'<div id="baz">Baz</div>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isTrue(actual);
+	});
+
+	it('returns false for `non-widget` role', function() {
+		var vNode = queryFixture(
+			'<a role="contentinfo" id="target" aria-label="some content">Content Information</a>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns true for non-widget role that does support name from content', function() {
+		var vNode = queryFixture(
+			'<div id="target" role="tooltip" aria-label="OK">Next</div>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isTrue(actual);
+	});
+
+	it('returns false for empty text content', function() {
+		var vNode = queryFixture(
+			'<button id="target" aria-label="close"></button>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns false for non text content', function() {
+		var vNode = queryFixture(
+			'<button id="target" aria-label="close"><i class="fa fa-icon-close"></i></button>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns false for hidden (non visible) text content', function() {
+		var vNode = queryFixture(
+			'<button id="target" aria-label="close"><span style="display:none"></span></button>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns true when visible text is combination of alphanumeric and emoji characters', function() {
+		var vNode = queryFixture(
+			'<button id="target" aria-label="I would like a burger">I would like a 🍔 </button>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isTrue(actual);
+	});
+
+	it('returns true when visible text is combination of alphanumeric and punctuation characters', function() {
+		var vNode = queryFixture(
+			'<button id="target" aria-label="next page">next page &gt;</button>'
+		);
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isTrue(actual);
+	});
+});
```

Comment:
You should add a test for an elm with an implicit name-from-content role that is replaced with an explicit role that does not have name from content. For example `<button role="tooltip">`

Reply:
`tooltip` does support name from content, for example `status` does not, added a testcase.